### PR TITLE
Rsvp

### PR DESCRIPTION
**Pull request by jeffjramos21**
Fixed rsvp to correctly update rsvp array when adding a new user to it. Added 404 error messages for when we try to access an event or user document that does not exist.

### CODE REVIEW

**Review by HanielDiaz — CHANGES_REQUESTED** [review]
Please download the linter and fix all the linting errors.

File: functions/src/Controller/Event.ts [hunk 1]
```diff
@@ -1,37 +1,158 @@
...
+	});
+
+	/**
+	 * If both documents exist, complete rsvp function.
+	 */
+	if (validEvent && validUser) {
+		rsvp(event, user);
+		response.status(200).send('Good Job');
+	}
+	else if (!validEvent && ! validUser) {
```
Comment (idelmr) [inline functions/src/Controller/Event.ts]: Remove space after `!`

File: functions/src/Controller/Event.ts [hunk 2]
```diff
@@ -1,37 +1,158 @@
...
+	});
+
+	/**
+	 * If both documents exist, complete checkIn function.
+	 */
+	if (validEvent && validUser) {
 		checkIn(event, user);
 		response.status(200).send('Good Job');
+	}
+	else if (!validEvent && ! validUser) {
```
Comment (idelmr) [inline functions/src/Controller/Event.ts]: Remove space after `!`

File: functions/src/Controller/Event.ts [hunk 3]
```diff
@@ -1,37 +1,158 @@
...
+	});
 });
 
-export const check_In = functions.https.onRequest((request, response) => {
-		const event: Event = new Event(request.body[0]);
-		const user: User = new User(request.body[1]);
+export const checkInController = functions.https.onRequest(async (request, response) => {
+	const event: Event = new Event(request.body[0]);
+	const user: User = new User(request.body[1]);
+
```
Comment (idelmr) [inline functions/src/Controller/Event.ts]: Remove empty  line

File: functions/src/Controller/Event.ts [hunk 4]
```diff
@@ -1,37 +1,158 @@
...
+	const eventCollection = getEventCollection();
+	const userCollection = getUserCollection();
+
```
Comment (idelmr) [inline functions/src/Controller/Event.ts]: Remove empty line

File: functions/src/Controller/Event.ts [hunk 5]
```diff
@@ -1,37 +1,158 @@
...
+	const eventDoc = eventCollection.doc(event.id);
+	const userDoc = userCollection.doc(user.id);
+
```
Comment (idelmr) [inline functions/src/Controller/Event.ts]: Remove line

File: functions/src/Controller/Event.ts [hunk 6]
```diff
@@ -1,37 +1,158 @@
...
+	});
 });
 
-export const rsvps = functions.https.onRequest((request, response) => {
+export const rsvpController = functions.https.onRequest(async (request, response) => {
 	const event: Event = new Event(request.body[0]);
 	const user: User = new User(request.body[1]);
-	rsvp(event, user);
-	response.status(200).send('Good Job');
+
```
Comment (idelmr) [inline functions/src/Controller/Event.ts]: Also remove empty lines between variable declarations

**Review by idelmr — CHANGES_REQUESTED** [review]
Minor changes

File: functions/src/Controller/Event.ts [hunk 7]
```diff
@@ -1,37 +1,152 @@
...
+	/**
+	 * If both documents exist, complete rsvp function.
+	 */
+	if (validEvent && validUser) {
+		rsvp(event, user);
+		response.status(200).send('Good Job');
+	}
+	else if (!validEvent && !validUser) {
+		response.status(200).send('Error 404: Event document not found\nError 404: User document not found');
```
Comment (HanielDiaz) [inline functions/src/Controller/Event.ts]: no. 404 is a status code. so don't send status 200. same for below

File: functions/src/Controller/Event.ts [hunk 8]
```diff
@@ -1,37 +1,152 @@
...
+
+	/**
+	 * If both documents exist, complete checkIn function.
+	 */
+	if (validEvent && validUser) {
 		checkIn(event, user);
 		response.status(200).send('Good Job');
+	}
+	else if (!validEvent && !validUser) {
+		response.status(200).send('Error 404: Event document not found\nError 404: User document not found');
```
Comment (HanielDiaz) [inline functions/src/Controller/Event.ts]: status should be 404 not 200 same for below

File: functions/src/Controller/Event.ts [hunk 9]
```diff
@@ -1,37 +1,152 @@
...
+	/**
+	 * If Event document exists, complete editEvent function.
+	 */
+	await eventDoc.get().then((docSnapshot) => {
+		if (docSnapshot.exists) {
+			editEvent(event);
+			response.status(200).send('Good Job');
+		}
+		else {
+			response.status(200).send('Error 404: Event document not found');
```
Comment (HanielDiaz) [inline functions/src/Controller/Event.ts]: status 404

**Review by HanielDiaz — CHANGES_REQUESTED** [review]

**Review by HanielDiaz — APPROVED** [review]
lgtm

**Review by idelmr — APPROVED** [review]

**Pull request merged**